### ISSUE TITLE
Comprehensive Resting Functionality

### ISSUE THREAD

**Issue by LukasPrism** [foundryvtt/dnd5e#3477]
As of recently, the invaluable Rest Recovery 5E module by Wasp is no longer being maintained. https://foundryvtt.com/packages/rest-recovery

For quite a while this module has filled the gaps in providing a robust, community guided, set of options and functionality around streamlining the resting process and how class features, feats, items affect and are affected by resting. It also includes food and water consumption tracking for those who like a bit more realism for dungeon/hex crawls.

I'm not sure how feasible it would be just to integrate this into dnd5e as it stands, perhaps even temporarily, to keep this functionality available with the fast-changing system. If not, at least it can maybe provide some guidance around what has proven popular and necessary in this regard – and now that it is potentially winding down that may affect your priorities.

**Comment by Rabulias** [comment]
I second this. The module got to be extensive in its functionality, though, so I don't know if it would be a good fit for _all_ of it to be incorporated into the DND5E system. Stuff like the Simple Calendar integration or Food & Water tracking might remain module territory. But basic HP recovery, Hit Dice recovery and expenditure, feature and spell recovery, item recharge, and exhaustion recovery would be most appreciated!

**Comment by Zanderaf** [comment]
Yea, being able to make features or way to restore x amount of spell slots on rest through some features, for various different classes is something that feels like it should be supported, Rest Recovery did it a bit, but stuff like Song of Rest, Arcane Recovery, Natural Recovery from Land Druid, etc

**Comment by Dyrnwyrm** [comment]
I agree with the previous exposed points, Rest Recovery fills a lot of gaps even letting you add some homebrew rules on it.
Item charges, food & water, feature and spell recharges are an important part of core that can even help setting a good base for future official adventures/modules

**Comment by A1exanderWebb** [comment]
I've had this module for so long that I forgot some of its features weren't part of core, but having the nice hit die ui during rests is something that makes rests in the middle of a session quick to perform and prevents them form derailing gameplay for more than a few moments.
Would be great to also see some form of incorporation of features like arcane recover or land harness divine power to restore spell slots or resources, even if it is nor necessarily part of the short/long rest workflow.

**Comment by TheDSeraph** [comment]
A great module that really should be folded into core!

**Comment by Sayshal** [comment]
Definitely a must for this system.

**Comment by ctbritt** [comment]
Please, please please please please incorporate this module into core!!!

**Comment by Fyorl** [comment]
We would want to add support for some existing rules that is currently lacking, but this issue is far too broad and needs to be broken down into individual feature requests.

**Comment by Kapuzenjoe** [comment]
One part of it would be the food and water tracker, for which there's already a separate request #1193
